drive/driveimpl/remote_impl.go

```diff
@@ -409,6 +415,18 @@ var writeMethods = map[string]bool{
 	"DELETE":    true,
 }
 
+// canSudo checks wether we can sudo -u the configured executable as the
+// configured user by attempting to call the executable with the '-h' flag to
+// print help.
+func (s *userServer) canSudo() bool {
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
+	defer cancel()
+	if err := exec.CommandContext(ctx, "sudo", "-n", "-u", s.username, s.executable, "-h").Run(); err != nil {
```

Comment:
:question: Non-blocking question: I always thought that `--validate` was the right way to check permissions, without actually needing to run the command. Does that fail?

Reply:
Is that a `sudo` flag. I don't see it [here](https://linux.die.net/man/8/sudo) for example.

Reply:
That’s might be an old manpage, where it only lists the short flag?
https://man7.org/linux/man-pages/man8/sudo.8.html
```
       -v, --validate
               Update the user's cached credentials, authenticating the
               user if necessary.  For the sudoers plugin, this extends
               the sudo, timeout for another 5 minutes by default, but
               does not run a command.  Not all security policies support
               cached credentials.
```